File: implementation/scipamato/scipamato-common-persistence-jooq/src/main/java/ch/difty/scipamato/persistence/config/JooqSpringBootConfiguration.java
package ch.difty.scipamato.persistence.config;

import org.jooq.conf.Settings;
import org.jooq.impl.DefaultConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class JooqSpringBootConfiguration {

    @Bean
    public Settings settings() {
        return new DefaultConfiguration().settings().withExecuteWithOptimisticLocking(true);
    }
}

File: common/common-persistence-jooq/src/main/kotlin/ch/difty/scipamato/common/persistence/config/JooqSpringBootConfiguration.kt
package ch.difty.scipamato.common.persistence.config

import org.jooq.conf.Settings
import org.jooq.impl.DefaultConfiguration
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class JooqSpringBootConfiguration {
    @Bean
    open fun settings(): Settings =
        DefaultConfiguration().settings()
            .withExecuteWithOptimisticLocking(true)
}
